crates/flux-middle/src/rty/expr.rs

```diff
@@ -620,6 +631,29 @@ pub enum UnOp {
     Neg,
 }
 
+#[derive(Copy, Clone, PartialEq, Eq, Hash, TyEncodable, TyDecodable)]
+pub enum Ctor {
+    /// for `refined_by` types
```

Comment:
I'd prefer to avoid the distinction `refined_by` vs `reflected` and use `struct` vs `enum`.

Reply:
Hmm not following here -- we also use `refined_by` for `enum` (e.g. for measures?)

Reply:
I want to be able to support the following

```rust
#[reflect]
struct Range<Idx> {
    pub start: Idx,
    pub end: Idx,
}
```

This should reflect the type as a struct, so it won't be true that reflected equals enum.

Reply:
Ok have tweaked the comment; still want to reflect (ha!) the fact that `enum` is only currently used for reflected things... [ `Struct` also means you do all those projection/packing shenanigans inside `flux` instead of leaving it to fixpoint/smt -- of course you _could_ leave it all to fixpoint for reflected-structs but that would be a bit of a change... ]